File: web-shop/src/app/services/auth.service.js
import HttpService from "./http-service";

class AuthService extends HttpService {
  async login(username, password) {
    try {
      const response = await this.post("/auth/login", {
        body: {
          username,
          password,
        },
      });

      return response.data;
    } catch (error) {
      alert("Đăng nhập lỗi")
      throw new Error(error.message);
    }
  }
  async signUp({ email, fullName, phone, username, address }) {
    try {
      const response = await this.post("/auth/signup", {
        body: {
          avatar: null,
          email,
          fullName,
          phone,
          username,
          address
        },
      });

      return response.data;
    } catch (error) {
      alert(error.response.data.message);
      throw new Error(error.message);
    }
  }
  async forgotPassword(email) {
    try {
      const response = await this.get("/auth/generate-token-pass", {
        params: {
          email,
        },
      });

      return response.data;
    } catch (error) {
      alert("Lấy mã xác nhận không thành công. Bạn vui lòng kiểm tra lại email đã nhập")
      throw new Error(error.message);
    }
  }
  async forgotPasswordConfirm({ username, newPassword, token }) {
    try {
      const response = await this.post("/auth/new-password", {
        body: {
          username,
          newPassword,
          token,
        },
      });

      return response.data;
    } catch (error) {
      alert("Không thành công")
      throw new Error(error.message);
    }
  }
  async changePassword({ username, password, newPassword }) {
    try {
      const response = await this.post("/auth/change-password", {
        body: {
          username,
          password,
          newPassword,
        },
      });

      return response.data;
    } catch (error) {
      throw new Error(error.message);
    }
  }
}

export default new AuthService();

File: beweb/src/main/java/com/project/beweb/service/AuthService.java
package com.project.beweb.service;

import com.project.beweb.dto.UserDTO;
import com.project.beweb.exception.LoginException;
import com.project.beweb.model.User;
import com.project.beweb.payload.request.AuthenticationRequest;
import com.project.beweb.payload.request.ChangePasswordRequest;
import com.project.beweb.payload.request.NewPasswordRequest;
import com.project.beweb.payload.response.AuthenticationResponse;
import com.project.beweb.payload.response.TrueFalseResponse;

import java.io.InvalidObjectException;

public interface AuthService {
    //Đăng nhập
    AuthenticationResponse login(AuthenticationRequest request) throws LoginException;

    //Kiểm tra token
    Boolean validateToken(AuthenticationResponse authenticationResponse);

    //Đăng ký tài khoản mới
    AuthenticationResponse signup(UserDTO userDTO) throws InvalidObjectException;

    //sét password mới sau khi đã có mã gửi về mail
    User setNewPassword(NewPasswordRequest newPasswordRequest);

    //Tạo mã gửi về mail
    TrueFalseResponse generateForgotTokenPass(String email);

    //Kiểm tra mã gửi về mail xem có hợp lệ không -> có thì gửi mail password mới
    User forgotPassword(String username, String token);

    TrueFalseResponse changePassword(ChangePasswordRequest request);
}
